src/components/activity-list/ActivityList.tsx

```diff
@@ -135,20 +130,42 @@ const ActivityList = lazyMount(() => {
     [nativeCurrency, theme]
   );
 
+  const listRef = useRef<LegendListRef | null>(null);
+
+  const scrollToTopRef = useMemo(() => {
+    return {
+      scrollToTop() {
+        if (!listRef.current) {
+          return;
+        }
+        if (listRef.current.getState().isAtStart) {
+          return;
+        }
+        listRef.current.scrollToIndex({
+          index: 0,
+          animated: true,
+        });
+      },
+    };
+  }, []);
+
+  useEffect(() => {
+    setScrollToTopRef?.(scrollToTopRef);
+  }, [scrollToTopRef, setScrollToTopRef]);
+
   return (
     <LegendList
       data={flatData}
-      ref={list => {
-        if (list) {
-          setScrollToTopRef?.({
-            scrollToTop() {
-              list.scrollToIndex({
-                index: 0,
-                animated: true,
-              });
-            },
-          });
-        }
+      // changing key - we had a bug with key calculation where headers were
+      // matching causing legend list to see the key move index and scroll to a
+      // bad position i tried fixing just the key to avoid changing the key
+      // here, but then legend list will not scroll to the top on switching
+      // wallets from home, in fact it seems to keep the scroll position down
+      // even if it was at the top before sometimes. so changing the key here
+      // ensures the list fully re-renders and has the correct scroll position
+      key={accountAddress}
+      ref={ref => {
+        listRef.current = ref;
       }}
```

Comment:
Inconsequential, but any reason to not just do `ref={listRef}` here?

Reply:
no i think just remnants from me moving over from using setState before, fixed